### ISSUE TITLE
Feature: divide functional and non-functional rendering

### ISSUE THREAD

**Issue by martinoppitz** [krausest/js-framework-benchmark#138]
Hello.

Angular JIT is non-functional rendering.
Angular AOT is functional rendering.

Vue actualy is non-functional rendering.
But there is a functional implementation possible.

Inferno is a functional rendering.

That makes a difference.

Some frameworks are fast but rather inappropriate. So you can use VanillaJS (@krausest).

**Comment by martinoppitz** [comment]
![image](https://cloud.githubusercontent.com/assets/6279703/23594584/532bd41a-021b-11e7-8568-c8a802b4a441.png)

**Comment by leeoniya** [comment]
the func-keyed results look suspect.

the fastest keyed "remove all rows" is kivi's 137ms. vanillajs is ~141ms. it's highly unlikely vue found a way to cut this down to just 73ms?

**Comment by martinoppitz** [comment]
*maybe non-keyed instead keyed*

**Comment by leeoniya** [comment]
yeah, probably. this benchmark uses mutationObservers to classify each impl into the proper table depending on how it behaves.

**Comment by krausest** [comment]
I need some background. What's the definition of "functional" and "non-functional" rendering? Using a template vs. using a javascript rendering function?

**Comment by martinoppitz** [comment]
Look at the table.component. the h function is like the same what angular aot does.

I think the frameworks shoud evaluate by features.

Components directives filter/pipes services maybe DI 

This features make us productive. And slow down the performance.

For example. With inferno I can't develop our current apps. I missing usefull features.

Sorry that my ideas

**Comment by krausest** [comment]
vue-func is not keyed according to my classification:
```
webdriver-ts stefan$ node dist/nonKeyed.js --framework vue-v2.2.1-func-keyed
Frameworks that will be checked [ 'vue-v2.2.1-func-keyed' ]
(node:39877) DeprecationWarning: os.tmpDir() is deprecated. Use os.tmpdir() instead.
vue-v2.2.1-func-keyed is non-keyed for 'run benchmark' and non-keyed for 'remove row benchmark' non-keyed for 'swap rows benchmark' . It'll appear as non-keyed in the results
ERROR: Framework vue-v2.2.1-func-keyed is not correctly categorized in commons.ts
```
Apart from that the "func" vue variant appears to be even slower for startup (65 msecs to 52 msecs).

The vue-keyed implementation already appears to translate the template to javascript like angular aot does (look at vue/dist/main.js). So I don't think anything can be gained by manually implementing the render function.
